### PR TITLE
Rianfowler/sc 52399/upgrade kots to react 18

### PR DESCRIPTION
**Pull request by rianfowler**
#### What this PR does / why we need it:
- upgrades kots to the latest version of react 
- gives us access to latest unit testing tools 
- react 18 has better optimized render performance 
- removes graphQL dependencies (unused) 

#### Which issue(s) this PR fixes:
https://app.shortcut.com/replicated/story/52399/upgrade-kots-to-react-18?cf_workflow=500129355&ct_workflow=all

#### Does this PR introduce a user-facing change?
```release-note
- upgrade kots to react 18
```

#### Does this PR require documentation?
NONE

### CODE REVIEW

File: web/src/components/UploadAirgapBundle.jsx [hunk 1]
```diff
@@ -36,7 +36,7 @@ class UploadAirgapBundle extends React.Component {
     uploadResuming: false,
   };
 
-  emptyHostnameErrMessage = "Please enter a value for \"Hostname\" field";
+  emptyHostnameErrMessage = 'Please enter a value for "Hostname" field';
```
Comment (rianfowler) [inline web/src/components/UploadAirgapBundle.jsx:39]: Looks like linting isn't running as part of CI. I will look into that

File: web/src/index.jsx [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
+const root = createRoot(container); // createRoot(container!) if you use TypeScript
 
-ReactDOM.render(
+root.render(
   <ReplicatedErrorBoundary>
     <Root />
-  </ReplicatedErrorBoundary>,
-  document.getElementById("app")
+  </ReplicatedErrorBoundary>
 );
```
Comment (rianfowler) [inline web/src/index.jsx:12]: These are the main changes. I followed this guide.

Comment (jodillion) [reply web/src/index.jsx:12]: @rianfowler can you link the guide?

**Review by jodillion — DISMISSED** [review]

File: web/src/index.jsx (hunk 2 above)
Comment (rianfowler) [reply web/src/index.jsx:12]: forgot to do that sorry https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html

**Review by miaawong — APPROVED** [review]

**Pull request closed without merging**